packages/components/ovh-shell/src/message-bus/iframe.ts

```diff
@@ -0,0 +1,47 @@
+import { IShellMessage } from '../common';
+import IMessageBus from './IMessageBus';
+
+const IFRAME_MESSAGE_TYPE = 'ovh-shell-iframe-message';
+
+export default class IFrameMessageBus implements IMessageBus {
+  iframe?: HTMLIFrameElement;
+
+  listeners: CallableFunction[];
+
+  constructor(iframe: HTMLIFrameElement = null) {
+    this.iframe = iframe;
+    this.listeners = [];
+    window.addEventListener('message', (event) => {
+      const { data } = event;
+      if (data.type === IFRAME_MESSAGE_TYPE) {
```

Comment:
**suggestion**:
Could the warning about event origin be addressed here? Or does it implies much complexity regarding communcating iframes?

Reply:
Fixed.